src/header.tsx

```diff
@@ -120,113 +135,98 @@ export const Header = () => {
-                <MenubarSeparator />
-                <MenubarItem>Print</MenubarItem>
-              </MenubarContent>
-            </MenubarMenu>
-          </Menubar>*/}
-          </div>
-          <div className="flex">
+        <div className="flex w-full justify-end">
+          <DropdownMenu>
+            <DropdownMenuTrigger asChild>
+              <Button
+                size="default"
+                variant={'outline'}
+                className="mx-2 cursor-pointer capitalize"
+              >
+                {network}
+                <ChevronDown className="ml-2 h-4 w-4" />
+              </Button>
+            </DropdownMenuTrigger>
+            <DropdownMenuContent>
+              {networkList.map(({ name, display }) => (
+                <DropdownMenuItem
+                  className="cursor-pointer"
+                  key={name}
+                  onClick={() => selectNetwork(name)}
+                >
+                  {display}
+                </DropdownMenuItem>
+              ))}
+            </DropdownMenuContent>
+          </DropdownMenu>
+          {selectedAccount?.address ? (
             <DropdownMenu>
               <DropdownMenuTrigger asChild>
                 <Button
+                  variant="outline"
                   size="default"
-                  variant={'outline'}
-                  className="mx-2 cursor-pointer capitalize"
+                  className="cursor-pointer overflow-hidden"
                 >
-                  {network}
+                  <Polkicon
+                    size={36}
+                    address={selectedAccount?.address || ''}
+                    className="mr-2"
+                    outerColor="transparent"
+                  />
+                  {selectedAccount?.name}
                   <ChevronDown className="ml-2 h-4 w-4" />
                 </Button>
               </DropdownMenuTrigger>
               <DropdownMenuContent>
-                {networkList.map(({ name, display }) => (
-                  <DropdownMenuItem
-                    className="cursor-pointer"
-                    key={name}
-                    onClick={() => selectNetwork(name)}
-                  >
-                    {display}
-                  </DropdownMenuItem>
+                {accounts.map((account, index) => (
+                  <>
+                    <DropdownMenuItem
+                      className="cursor-pointer"
+                      key={account.address}
+                      onClick={() => selectAccount(account)}
+                    >
+                      <Polkicon
+                        size={28}
+                        address={account.address || ''}
+                        className="mr-2"
+                        outerColor="transparent"
+                      />
+                      {account.name}
+                    </DropdownMenuItem>
+                    {index !== accounts.length - 1 && <DropdownMenuSeparator />}
```

Comment:
I think we need a key for this separator, we should check if there's an error there. I've see it in the logs in the past days.
```suggestion
                    {index !== accounts.length - 1 && <DropdownMenuSeparator key={`${account.address}-separator`} />}
```